src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java

```diff
@@ -1,169 +1,218 @@
+        this.homeOfficeApi = homeOfficeApi;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCaseWithHomeOfficeData =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        AppealType appealType = asylumCaseWithHomeOfficeData.read(APPEAL_TYPE, AppealType.class)
+                .orElseThrow(() -> new IllegalStateException("AppealType is not present."));
+
+        if (!HomeOfficeAppealTypeChecker.isAppealTypeEnabled(featureToggler, appealType)) {
+
+            return new PreSubmitCallbackResponse<>(asylumCaseWithHomeOfficeData);
+        }
+
+        final String homeOfficeSearchStatus = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_SEARCH_STATUS, String.class)
+            .orElse("");
+        final YesOrNo homeOfficeNotificationsEligible
+            = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_NOTIFICATIONS_ELIGIBLE, YesOrNo.class)
+            .orElse(YesOrNo.NO);
+        final long caseId = callback.getCaseDetails().getId();
+        final String homeOfficeReferenceNumber
+            = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse("");
+
+        if (asylumCaseWithHomeOfficeData.read(APPELLANT_IN_UK, YesOrNo.class).map(
+            value -> value.equals(YesOrNo.YES)).orElse(true)) {
+
+            // RIA-5683: Prevent multiple notifications for the same event and case ID.
+            if (Objects.equals(cachedEvent, callback.getEvent().toString()) && cacheCaseId == caseId) {
+                log.info("Home Office notification already invoked: "
+                         + SUPPRESSION_LOG_FIELDS,
+                        callback.getEvent(), caseId, homeOfficeReferenceNumber, homeOfficeSearchStatus,
+                        homeOfficeNotificationsEligible);
+                return new PreSubmitCallbackResponse<>(asylumCaseWithHomeOfficeData);
+            }
+            if ("SUCCESS".equalsIgnoreCase(homeOfficeSearchStatus)
+                && homeOfficeNotificationsEligible == YesOrNo.YES) {
+
+                log.info("Start: Sending Home Office notification - " + SUPPRESSION_LOG_FIELDS,
+                    callback.getEvent(), caseId, homeOfficeReferenceNumber, homeOfficeSearchStatus,
+                    homeOfficeNotificationsEligible);
+
+                asylumCaseWithHomeOfficeData = homeOfficeApi.aboutToSubmit(callback);
+
+                log.info("Finish: Sending Home Office notification - " + SUPPRESSION_LOG_FIELDS,
+                    callback.getEvent(), caseId, homeOfficeReferenceNumber, homeOfficeSearchStatus,
+                    homeOfficeNotificationsEligible);
+                // We store the case Id to prevent duplicated notifications.

```

Comment:
Update the comment to clarify that we are caching event and id to prevent duplicated notifications.